src/consent.ts

```diff
@@ -0,0 +1,104 @@
+import { PostHog } from './posthog-core'
+import { find, includes } from './utils'
+import { assignableWindow, navigator } from './utils/globals'
+import { cookieStore, localStore } from './storage'
+import { PersistentStore } from './types'
+
+const OPT_OUT_PREFIX = '__ph_opt_in_out_'
+
+export enum ConsentStatus {
+    PENDING = -1,
+    DENIED = 0,
+    GRANTED = 1,
+}
+
+/**
+ * ConsentManager provides tools for managing user consent as configured by the application.
+ */
+export class ConsentManager {
+    private _storage?: PersistentStore
+
+    constructor(private instance: PostHog) {}
+
+    private get config() {
+        return this.instance.config
+    }
+
+    public get consent(): ConsentStatus {
+        if (this.getDnt()) {
+            return ConsentStatus.DENIED
+        }
+
+        return this.storedConsent
+    }
+
+    public isOptedOut() {
+        return (
+            this.consent === ConsentStatus.DENIED ||
+            (this.consent === ConsentStatus.PENDING && this.config.opt_out_capturing_by_default)
+        )
+    }
+
+    public isOptedIn() {
+        return !this.isOptedOut()
+    }
+
+    public optInOut(isOptedIn: boolean) {
+        this.storage.set(
+            this.storageKey,
+            isOptedIn ? 1 : 0,
+            this.config.cookie_expiration,
+            this.config.cross_subdomain_cookie,
+            this.config.secure_cookie
+        )
+    }
+
+    public reset() {
+        this.storage.remove(this.storageKey, this.config.cross_subdomain_cookie)
+    }
+
+    private get storageKey() {
+        const { token, opt_out_capturing_cookie_prefix } = this.instance.config
+        return (opt_out_capturing_cookie_prefix || OPT_OUT_PREFIX) + token
+    }
+
+    private get storedConsent(): ConsentStatus {
+        const value = this.storage.get(this.storageKey)
+        return value === '1' ? ConsentStatus.GRANTED : value === '0' ? ConsentStatus.DENIED : ConsentStatus.PENDING
```

Comment:
```suggestion
        return ConsentStatus[value] || ConsentStatus.PENDING
```

I _think_ that works

Reply:
arguable that your more explicit version is more clear though so 🤷

Reply:
It's also _not_ the same I think. The stored value is a string so we would have to parse it to a number

Reply:
Ah, yeah, I missed that... make the enum values strings 🙈 

:)